src/modules/launcher/Plugins/Microsoft.Plugin.Program/Storage/Win32ProgramRepository.cs

```diff
@@ -138,11 +141,11 @@ private void OnAppDeleted(object sender, FileSystemEventArgs e)
         }
 
         // When a URL application is deleted, we can no longer get the HashCode directly from the path because the FullPath a Url app is the URL obtained from reading the file
-        private Win32 GetAppWithSameNameAndExecutable(string name, string executableName)
+        private Win32Program GetAppWithSameNameAndExecutable(string name, string executableName)
         {
-            foreach (Win32 app in Items)
+            foreach (Win32Program app in Items)
             {
-                if (name.Equals(app.Name, StringComparison.OrdinalIgnoreCase) && executableName.Equals(app.ExecutableName, StringComparison.OrdinalIgnoreCase))
+                if (name.Equals(app.Name, StringComparison.CurrentCultureIgnoreCase) && executableName.Equals(app.ExecutableName, StringComparison.CurrentCultureIgnoreCase))
```

Comment:
When would it be better to use `CurrentCultureIgnoreCase` and `InvariantCultureIgnoreCase`?

Reply:
A good description can be found [here](https://docs.microsoft.com/en-us/dotnet/standard/base-types/best-practices-strings).  My logic here was that application names could be dependent on the current culture hence the change.    In most cases Ordinal should be used instead of the Invariant Culture except when persisting linguistically meaningful information in a culturally agnostic way as described [here](https://docs.microsoft.com/en-us/dotnet/standard/globalization-localization/performing-culture-insensitive-string-comparisons).   Example sorting strings consistently across cultures from multiple character sets.